### PR TITLE
gh-112088: Run autoreconf in GHA check_generated_files

### PR DESCRIPTION
**Pull request by vstinner**
The "Check if generated files are up to date" job of GitHub Actions now runs the "autoreconf -ivf -Werror" command instead of the "make regen-configure" command to avoid depending on the unstable quay.io server.

* Issue: gh-112088

### CODE REVIEW

**Review by gpshead — APPROVED** [review]
I think this is a good idea. It doesn't appear there is any reason for the quay.io image anymore given ubuntu-22.04 has a sufficiently good autoconf.

**Comment by vstinner** [comment]
cc @erlend-aasland

File: Makefile.pre.in [hunk 1]
```diff
@@ -2644,12 +2644,7 @@ autoconf:
 # See https://github.com/tiran/cpython_autoconf container
```
Comment (hugovk) [inline Makefile.pre.in]: Do we need this link to the old container repo? 
```suggestion
```

**Review by hugovk — COMMENTED** [review]
Let's update https://github.com/python/cpython/blob/d4f83e1e3a19e2f881115f20d58ae6a019ddb48f/Doc/using/configure.rst#generated-files as well.

And backport?

**Comment by vstinner** [comment]
@hugovk: I completed the PR to address your review. Would you mind to review the updated PR?

> Let's update https://github.com/python/cpython/blob/d4f83e1e3a19e2f881115f20d58ae6a019ddb48f/Doc/using/configure.rst#generated-files as well.

Done.

> And backport?

I added the backport to 3.11 and 3.12 labels.

**Comment by vstinner** [comment]
The main point of this PR is that `Tools/build/regen-configure.sh` now uses the same container image than the GitHub Action job, so the job doesn't have to run `autoreconf` in a container, but can run it directly.

**Comment by vstinner** [comment]
Oh, there is a fix in the Changelog entry causing the Docs job to fail: fixed.

**Review by hugovk — APPROVED** [review]

**Comment by vstinner** [comment]
@erlend-aasland: It would be nice if you could review this change, since you made many configure changes last months ;-)

**Comment by erlend-aasland** [comment]
Also, the devguide must be updated.

**Review by erlend-aasland — APPROVED** [review]
This looks good to me. (I'm a little bit puzzled by the `aclocal.m4` and `configure` changes, but they are also fine.)

Thanks for doing this.

**Comment by erlend-aasland** [comment]
IIRC, we did the autoconf version bump before the 3.12 freeze, so IMO we should backport this to 3.12.

**Comment by vstinner** [comment]
> This looks good to me. (I'm a little bit puzzled by the aclocal.m4 and configure changes, but they are also fine.)

I'm not sure if I picked the wrong `pkg-config` package. But well, for now, my main worry is more to make the whole Python workflow more reliable by avoiding depending on the external `quay.io` server which had multiple outages in several days.

With my change, the Python workflow no longer pulls an external container image, but simply reuse what's available on Ubuntu 22.04 in the GitHub Action.

> IIRC, we did the autoconf version bump before the 3.12 freeze, so IMO we should backport this to 3.12.

Oh. So we cannot backport this change to 3.11?

**Comment by vstinner** [comment]
Merged. Thanks for reviews. Let's start with a backport to 3.12.

**Comment by erlend-aasland** [comment]
> Oh. So we cannot backport this change to 3.11?

No, 3.11 uses a patched Autoconf 2.69 from Alpine Linux, not a vanilla 2.69 installation.

**Comment by sethmlarson** [comment]
@vstinner I was doing some work on release-tools adjacent to this, @zware pointed me to this PR. What are your thoughts on backporting the `regen-configure` makefile target to all supported versions so that release-tools can depend on it being there?

**Comment by gpshead** [comment]
For 3.11 and earlier autoconf version configure generation udpates, ask the release managers (@pablogsal and @ambv) for those branches.  (it makes sense for 3.12)

**Comment by efimov-mikhail** [comment]
I can see using of `aclocal 1.16.5` in this PR.
But in docs we write "Autoconf 2.71 and aclocal 1.16.4 are required to regenerate the configure script". (https://docs.python.org/3.14/using/configure.html#build-requirements).

Should we provide some changes to docs for update them?
@vstinner

**Comment by vstinner** [comment]
> Should we provide some changes to docs for update them?

Yes. Do you want to propose a PR to update the doc?

**Pull request merged**